src/database/DataTypes/StocktakeItem.js

```diff
@@ -305,7 +305,16 @@ export class StocktakeItem extends Realm.Object {
    */
   get hasReason() {
     if (!this.batches) return false;
-    return this.batches.some(batch => batch.option);
+    return this.batches.every(batch => {
+      if (batch.snapshotTotalQuantity !== batch.countedTotalQuantity) {
+        return !!batch.option;
+      }
+      return true;
+    });
+  }
+
+  get hasAnyReason() {
+    return this.batches.some(({ option }) => !!option);
```

Comment:
```suggestion
    return this.batches.some(batch => !!batch.option);
```
I personally find this clearer, but maybe I'm just behind the times. It just reads slightly more obviously to me that it's "in `batches` some `batch` has an option".

Reply:
can I disagree with this and just not do it?

Reply:
Yup totally